domain/storageprovisioning/state/state.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/machine"
+	"github.com/juju/juju/domain"
+	domainlife "github.com/juju/juju/domain/life"
+	machineerrors "github.com/juju/juju/domain/machine/errors"
+	"github.com/juju/juju/internal/errors"
+)
+
+type State struct {
+	*domain.StateBase
+}
+
+// CheckMachineIsDead checks to see if a machine is not dead returning
+// true when the life of the machine is dead.
+//
+// The following errors may be returned:
+// - [github.com/juju/juju/domain/machine/errors.MachineNotFound] when no
+// machine exists for the provided uuid.
+func (st *State) CheckMachineIsDead(
+	ctx context.Context, uuid machine.UUID,
+) (bool, error) {
+	db, err := st.DB()
+	if err != nil {
+		return false, errors.Capture(err)
+	}
+
+	var (
+		input       = machineUUID{UUID: uuid.String()}
+		machineLife machineLife
+	)
+	stmt, err := st.Prepare(
+		"SELECT $machineLife.* FROM machine WHERE uuid = machineUUIDVal.uuid",
```

Comment:
```
`
SELECT $machineLife.*
FROM    machine
WHERE   uuid = machineUUIDVal.uuid`,
```

Reply:
It doesn't need to be, what is the justification?

Reply:
Readability.